dependencies/kyverno/policy/pod-requests-del.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: remove-tekton-taskrun-resource-requests
+spec:
+  rules:
+  - name: remove-resource-requests-for-tekton-taskrun-pods
+    match:
+      any:
+      - resources:
+          kinds:
+          - Pod
+          selector:
+            matchLabels:
+              tekton.dev/task: verify-enterprise-contract
+    mutate:
+      foreach:
+      - list: "request.object.spec.containers"
+        patchesJson6902: |-
+          - op: replace
+            path: /spec/containers/{{elementIndex}}/resources/requests
+            value:
+              memory: 1Mi
+              cpu: 1m
```

Comment:
I'm wondering if this can have an impact on the performance of the build pipeline.

When I look at the duration of some recent test runs, they take [about 30 minutes](https://github.com/konflux-ci/konflux-ci/actions/runs/14239443029), while the sanity test in this PR took [~10 minutes more](https://github.com/konflux-ci/konflux-ci/actions/runs/14219433439), mostly caused by the build pipeline duration.

Would it be worth a try of adding a higher value here?

Reply:
With the label used, the only task that should be affected is the ec task. The issue with this task is that it has multiple steps (7 or 10 - don't recall) so whatever we're setting is multipled by that number. 
I'll run the ci a few times more to see if the behavior you mention is consistent. If it dies, maybe we'll to make some adjustments.

Reply:
Does the build pipeline duration takes into account the ec task run as well, running after the pipeline finishes?
Otherwise, it should not be affected by the change.

Reply:
ah, good point, I'm blind and totally missed `tekton.dev/task: verify-enterprise-contract`

Then it's very strange, because the EC task is a part of another test case, separated from the build pipeline test case.

Then the other explanation of this behavior could be a slower network? I.e. it took unusually long time to pull the images related to the build pipeline?

Reply:
I'll run it a few more times. Sometimes things take longer. Hopefully this was one such time. 😀

Generally speaking, I think we can craft more elaborated workarounds with this approach. I was hoping for a one-off solution for this issue, but seems as some tasks do need resource requests. At least it should allow us a way to address different tasks in different ways, and in a way that will apply to both ci and user envs.

Reply:
It seems to take longer consistently. Need to figure out why.

Reply:
Might be that having an admission webhook on taskrun has some overhead which affects the build stage the most as it has many taskruns (although untouched)

Reply:
I looked at the artifacts -> TaskRuns and found that the bottleneck is `clair-scan` Task:
```
    - container: step-get-vulnerabilities
      imageID: quay.io/konflux-ci/clair-in-ci@sha256:4227b14f54419af4dd42ff43d5e26bc9e19a52aaedc6a3d3c71b0b8213122953
      name: get-vulnerabilities
      terminated:
        containerID: containerd://3996f3bb55984faf4ac00ce1dc08dd2c0baa129664c85a3373041c6f92564231
        exitCode: 0
        finishedAt: "2025-04-03T11:53:57Z"
        reason: Completed
        startedAt: "2025-04-03T11:32:42Z"
      terminationReason: Completed
```

Reply:
🤦‍♂️
Not sure if good news or bad 😅

Reply:
Hmm on the other hand I found [the test run when the build pipeline took almost 1 hour to complete](https://github.com/konflux-ci/konflux-ci/actions/runs/14216658484/job/39834852270) 😮 

So good news is that this might not be related to your changes, bad news is that this sometimes happens for (to me) an unknown reason 😅

Reply:
Of all the recent PR runs I checked, the runs for this PR certainly appear to have taken longer than others.

Reply:
I was thinking maybe it takes too long for Kyverno to process the admission requests (or whatever it's called) when pods are created, and as the build process has many pods (task pods specifically), maybe that's the reason we see performance issues.
I now removed the customizations I did for reducing the resource requests for the controller pods and I think the results are better.

Reply:
it really looks much better now, thanks!

<img width="612" alt="Screenshot 2025-04-04 at 13 28 42" src="https://github.com/user-attachments/assets/f8c27a10-38b6-48bf-83c5-d65da34f46e4" />